src/middleware/navigation/check.condition.ts

```diff
@@ -68,6 +68,10 @@ export const checkOverseasEntityNumberEntered = (appData: ApplicationData): bool
   return checkHasAppData(appData) && (appData[OeNumberKey] || "").length !== 0;
 };
 
+export const checkUpdatePresenterEntered = (appData: ApplicationData): boolean => {
+  return checkOverseasEntityNumberEntered(appData) && Object.keys(appData[PresenterKey] || {}).length !== 0;
```

Comment:
Should this not check  checkHasAppData(appData) && ...?

Reply:
The `checkOverseasEntityNumberEntered` does that as its first condition.

```
export const checkOverseasEntityNumberEntered = (appData: ApplicationData): boolean => {
  return checkHasAppData(appData) && (appData[OeNumberKey] || "").length !== 0;
};
```

Reply:
But why check checkOverseasEntityNumberEntered?

Reply:
@AndreKramer7654 , this is checking OeNumberKey, as that is what is stored in the session after a successful search from overseas-entity-query page